stories/02-features/05-Style-with-themes.stories.mdx

```diff
@@ -0,0 +1,216 @@
+
+const MyComponent = (props) => {
+  const { theme } = props;
+
+  return (
+    <div className={theme.wrapper}>
+      <span className={theme.content}>This is the component content</span>
+    </div>
+  )
+}
+
+export default withThemes(
+  'component-identifier',
+  {
+    default: defaultTheme,
+    custom: customTheme,
+  },
+  true // composes
+)(MyComponent);
+```
+
+As illustrated above, the `withThemes` HOC requires:
+* A unique identifier. This is used for pulling a component's themes out of a context provider.
+* An object map of the component's available themes.
+* An optional `composes` boolean flag. If this is turned on, theme classes will be _added_ to default classes instead of replacing them, effectively using the default theme as a basis for all other themes.
+
+This HOC will provide the following functionality to the component on which it is applied:
+* A `theme` prop containing the classNames of the current theme (either `composed` or not, depending on if the flag is set).
+* Allow the `theme` prop to be derived in one of three ways:
+  * Pulled out of the component's predefined theme mapping using a `themeName` prop to access the appropriate key in the mapping object.
+  * Pulled out of React Context using the `indentifier` supplied to the HOC to access the necessary `themeName`, which will be in turn used to access the `theme` via the theme mapping object.
+  * Supplied directly to the component via a `theme` prop.
+
+Also illustrated above you can see the `className` values on the theme itself are accessed like an object, just as you would with css modules. Note that if a theme does not supply a `className` to override every class used in the component, the `className` value from the default theme will be used. The markup in the above example should look something like this (note that `composes` is set to true in this example):
+
+```html
+<div class="default__wrapper__1n274 customTheme__wrapper__n38f7">
+  <span class="default__content__v7892 customTheme__content__dkvus">This is the component content</span>
+</div>
+```
+
+That's it! As you can see, theming with Irving Core is relatively similar to theming with `styled-components`. Happy styling!
+
+## Additional notes on providing themes via WP-Irving
+
+Extending the styled-component theming plan, themes can also be applied to components in [WP Irving](https://github.com/alleyinteractive/wp-irving), the WordPress plug in to Irving. Components like post-list, site-menu, social sharing, etc., can be themed by determining the styled-component **alias** for that component and theming that component.
+
+For example `irving/post-title` is an alias for the `irving/text` component and can be themed via that component. You may apply a theme that already exists, like `h1`, or one you make a custom theme like `special-post-title`. If it's not obvious what the alias might be, use the WordPress editor to check the component's API and you'll see something like this:
```

Comment:
> If it's not obvious what the alias might be, use the WordPress editor to check the component's API and you'll see something like this:

We can also point to http://storybook.irvingjs.com/?path=/docs/wordpress-components-table-of-contents--page

Each of those components documents what the alias is

Reply:
Ah yeah, this is better